sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -67,9 +77,14 @@ class SimpleFunctionRegistry extends FunctionRegistry {
   }
 
   override def lookupFunction(name: String, children: Seq[Expression]): Expression = {
+    val builder = functionBuilders.get(name)
+    if (builder.isEmpty) {
+      throw new AnalysisException(s"undefined function $name")
+    }
     val func = synchronized {
-      functionBuilders.get(name).map(_._2).getOrElse {
-        throw new AnalysisException(s"undefined function $name")
+      Try(builder.map(_._2)) match {
```

Comment:
Hive's exception will be thrown. I faced the exception thrown in working on the changes. One jenkins test checks the error message. We will miss the actual exception and message and only get AnalysisException with `undefined function` here. I think it is proper to show real error message.